SW-1224 support legacy image for new update mechanism

JFSolution opened the pull request:
- override version of a dependency file from the cloud config
- add comparision option to the beamos_version of the cloud config
- refactor beamos_date to beamos_version

Commit:
refactor beamos_date to beamos_version

Commit:
refactor config generation to use beamos_version instead of beamos_date
change the testcases to do the same

Commit:
add comparison option to the update config for a beamos version

Commit:
added missing change

Commit:
change packaging version to pkg_resources as this is already on our devices

Commit:
fix dependency version for archive
removed python-version-warning as this is incompatible to the legacy image

Commit:
remove unused method

JFSolution commented:
the changes of the PR https://github.com/mrbeam/MrBeamPlugin/pull/1456 are also included in this PR as it wasn't merged so far

Commit:
added feedback of PR

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -479,35 +482,95 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
+            comparision_string (str): identifier to search for
+            comparision_options (list:
+
+        Returns:
```
missing return in doc string

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -479,35 +482,95 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
+        tierversion (str): software tier
 
     Returns:
         beamos config of the tierversion
```
here, Please mention type then description of the return  ex. 
`(dict): beamos config of the tierversion`

Commit:
added PR feedback
https://github.com/mrbeam/MrBeamPlugin/pull/1458#discussion_r850216475
https://github.com/mrbeam/MrBeamPlugin/pull/1458#discussion_r850215541
